### PR TITLE
adding py-pyspy package

### PR DESCRIPTION
**Pull request by lee218llnl**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage and linux sha256.

**Comment by tldahlgren** [comment]
Looks like `isort` isn't happy with the sort order:

```
...
==> Running isort checks
--- /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-pyspy/package.py:before	2023-11-14 01:11:49.966635
+++ /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-pyspy/package.py:after	2023-11-14 01:12:06.442600
@@ -3,8 +3,9 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import sys
+
 from spack.package import *
-import sys
 
 
 class PyPyspy(PythonPackage):
ERROR: /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-pyspy/package.py Imports are incorrectly sorted and/or formatted.
  isort found errors
...
```

**Comment by lee218llnl** [comment]
modified to import sys first

**Comment by tldahlgren** [comment]
Almost there, there's a missing blank line:

```
==> Running isort checks
--- /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-pyspy/package.py:before	2023-11-14 01:27:48.723399
+++ /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-pyspy/package.py:after	2023-11-14 01:28:03.342201
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import sys
+
 from spack.package import *
 
 
ERROR: /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-pyspy/package.py Imports are incorrectly sorted and/or formatted.
  isort found errors
  ```

File: var/spack/repos/builtin/packages/py-pyspy/package.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import sys
+
+from spack.package import *
+
+
+class PyPyspy(PythonPackage):
+    """py-spy is a sampling profiler for Python programs."""
+
+    homepage = "https://github.com/benfred/py-spy"
+    url = "https://files.pythonhosted.org/packages/51/70/e8bfeeac2c57f3af9988ca6d6d12cff3a9a1394e0d9a2c63f08555c0552f/py_spy-0.1.0-py2.py3-none-manylinux1_x86_64.whl"
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-pyspy/package.py:15]: We don't add wheels to Spack but try to build from source if possible, see <https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#downloading>.
Can you change that?

**Comment by lee218llnl** [comment]
turns out there is already a py-py-spy package https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/py-py-spy/package.py that does a proper cargo build from source.

**Pull request closed without merging**